server/aesdsocket.c

```diff
@@ -0,0 +1,200 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+#include <syslog.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#define BUFFER_SIZE 		  100
+#define PORT_NUMBER 		"9000"
+#define PATH_TO_FILE 		"/var/tmp/aesdsocketdata.txt"
+#define MAX_PENDING_CONNECTIONS  10
+
+int sock_fd,conn_fd; 
+
+struct addrinfo hints;
+struct addrinfo *servinfo;
+struct sockaddr_in clt_addr; 
+
+char *h_dynamic_buff = NULL;
+
+void signal_handler(int signal_t)
+{
+	if(signal_t==SIGINT){
+		syslog(LOG_INFO,"Ctrl+C Signal. Terminating\n");
+	}
+	else if(signal_t==SIGTERM){
+		syslog(LOG_INFO,"SIGTERM received. Gracefully terminating\n");
+	}
+	else{
+		syslog(LOG_INFO,"Received Unanticipated signal %d\n", signal_t);
+		exit(EXIT_FAILURE);
+	}
+	unlink(PATH_TO_FILE);
+	//Close socket and client connection
+	close(sock_fd);
+	close(conn_fd);
+	
+	exit(0); //Exit success 
+}
+
+
+int main(int argc, char *argv[])
+{ 
+	char input_buffer[BUFFER_SIZE]; 		//buffer to store the packets byte
+	bool packet_received=false;
+	int file_fd =0; 
+	int packet_bytes =0;
+	int total_packets =0; 
+	int rec_status =0, i =0;
+	socklen_t address_len=sizeof(struct sockaddr); 
+	
+	openlog(NULL,0, LOG_USER); 			//To setup logging with LOG_USER
+	
+	//To start a daemon process
+	if((argc>1) && strcmp(argv[1],"-d")==0)
+	{
+		if(daemon(0,0)==-1) {
+			syslog(LOG_ERR, "Unable to Enter Daemon Mode\n");
+			exit(1);
+		}
+	}
+	if(signal(SIGINT,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGINT failed");
+		exit(EXIT_FAILURE);
+	}
+	if(signal(SIGTERM,signal_handler)==SIG_ERR)
+	{
+		syslog(LOG_ERR,"SIGTERM failed");
+		exit(EXIT_FAILURE);
+	}
+	
+	//create socket for IPC
+	sock_fd=socket(PF_INET, SOCK_STREAM, 0);
+	if(sock_fd==-1) {
+		syslog(LOG_ERR, "Unable to create asocket\n");
+		exit(EXIT_FAILURE);
+	}
+	//Get server address
+	hints.ai_flags=AI_PASSIVE; 
+	if(getaddrinfo(NULL,PORT_NUMBER,&hints,&servinfo) !=0) {
+		syslog(LOG_ERR, "Unable to get the server's address\n");
+		exit(EXIT_FAILURE);
+	}
+	
+	//Bind the socket 
+	if(bind(sock_fd,servinfo->ai_addr,sizeof(struct sockaddr)) == -1) {
+		syslog(LOG_ERR, "Unable to Bind\n");
+		freeaddrinfo(servinfo); 			//Freeing the memory created by socket address before exiting 
+		exit(EXIT_FAILURE);
+	}
+	freeaddrinfo(servinfo); 				//Freeing the memory created by socket address
+	
+	file_fd=open(PATH_TO_FILE, O_WRONLY|O_CREAT|O_TRUNC, S_IRWXU|S_IRGRP|S_IROTH); 	//Creating a new file with 744 octal value
+	if (file_fd==-1) {
+		syslog(LOG_ERR, "Unable to create the file\n");
+		exit(EXIT_FAILURE);
+	}
+	close(file_fd);
+
+	//Looping for all the connections from Client
+	while(1)
+	{
+		if(listen(sock_fd,MAX_PENDING_CONNECTIONS) == -1) {		//Will start rejecting after 32 pending connections 
```

Comment:
@harshberiwal you shoudn't need `listen` in your while loop and at least one student noticed issues with this - see https://discord.com/channels/1062535844696707115/1079159844252176425/1079209265857835069